application/frontend/controllers/ResetController.php

```diff
@@ -81,9 +128,98 @@ public function actionResend($uid)
             throw new NotFoundHttpException();
         }
 
+        /*
+         * Resend verification
+         */
         $reset->send();
 
-        \Yii::$app->response->statusCode = 204;
-        return;
+        return $reset;
+    }
+
+    /**
+     * Validate reset code. Logs user in if successful
+     * @param string $uid
+     * @return Reset
+     * @throws BadRequestHttpException
+     * @throws NotFoundHttpException
+     * @throws ServerErrorHttpException
+     * @throws \Exception
+     */
+    public function actionValidate($uid)
+    {
+        /** @var Reset $reset */
+        $reset = Reset::findOne(['uid' => $uid]);
+        if ($reset === null) {
+            throw new NotFoundHttpException();
+        }
+
+        $code = \Yii::$app->request->getBodyParam('code', null);
+        if ($code === null) {
+            throw new BadRequestHttpException('Code is required', 1462989866);
+        }
+
+        $log = [
+            'action' => 'Validate reset',
+            'reset_id' => $reset->id,
+        ];
+
+        $isValid = $reset->isUserProvidedCodeCorrect($code);
+        if ($isValid === true) {
+
+            EventLog::log(
+                'ResetVerificationSuccessful',
+                [
+                    'reset_id' => $reset->id,
+                    'type' => $reset->type,
+                    'attempts' => $reset->attempts,
+                ],
+                $reset->user_id
+            );
+
+            /*
+             * Reset verified successfully, log user in
+             */
+            if (\Yii::$app->user->login($reset->user, \Yii::$app->params['sessionDuration'])) {
```

Comment:
I'm not sure the second parameter to `yii\web\User->login(...)` actually affects the session lifetime. Per [the documentation](http://www.yiiframework.com/doc-2.0/yii-web-user.html#login%28%29-detail), it appears to simply add the ability to use a cookie to maintain the user's logged-in status, which I don't think is what we want.

It looks like `yii\web\User->authTimeout` is where we should be putting the value we're currently putting into our `params` under `'sessionDuration'`.